src/elements/content-sidebar/activity-feed/annotation-thread/useReplies.js

```diff
@@ -0,0 +1,147 @@
+// @flow
+import React from 'react';
+import uniqueId from 'lodash/uniqueId';
+import APIFactory from '../../../../api';
+import commonMessages from '../../../common/messages';
+import { repliesErrors } from './errors';
+
+import type { BoxItemPermission, User } from '../../../../common/types/core';
+import type { BoxCommentPermission, Comment, FeedItemStatus } from '../../../../common/types/feed';
+import type { ElementsXhrError } from '../../../../common/types/api';
+
+type Props = {
+    annotationId: string,
+    api: APIFactory,
+    currentUser: User,
+    fileId: string,
+    filePermissions: BoxItemPermission,
+    initialReplies: Array<Comment>,
+};
+const useReplies = ({ annotationId, api, initialReplies = [], currentUser, fileId, filePermissions }: Props) => {
+    const [replies: Array<Comment>, setReplies] = React.useState(initialReplies);
+
+    React.useEffect(() => {
+        setReplies(initialReplies);
+    }, [initialReplies]);
+
+    const updateReplyItem = (updatedReplyValues: Object, replyId: string) => {
+        if (!replies) {
+            return;
+        }
+        const updatedReplies: Array<Comment> = replies.map(reply => {
+            if (reply.id === replyId) {
+                return {
+                    ...reply,
+                    ...updatedReplyValues,
+                };
+            }
+            return reply;
+        });
+
+        setReplies(updatedReplies);
+    };
+
+    const setReplyPendingStatus = (replyId: string, pendingStatus: boolean) => {
+        updateReplyItem({ isPending: pendingStatus }, replyId);
+    };
+
+    const addNewPendingReply = (baseComment: Object) => {
+        const date = new Date().toISOString();
+        const pendingReply: Comment = {
+            created_at: date,
+            created_by: currentUser,
+            modified_at: date,
+            isPending: true,
+            ...baseComment,
+        };
+
+        setReplies([...replies, pendingReply]);
+    };
+
+    const createReplySuccessCallback = (replyId: string, comment: Comment) => {
+        updateReplyItem(
+            {
+                ...comment,
```

Comment:
We're mixing "reply" and "comment" quite a lot in different areas, which could be confusing to the next maintainer. Can we draw a line of demarcation where one becomes the other, naming-wise?